frontend/server/src/Controllers/User.php

```diff
@@ -3086,57 +3110,86 @@ public static function getCoderOfTheMonthDetailsForSmarty(
     /**
      * @return array{smartyProperties: array{profile: array{userinfo: array{birth_date: null|string, classname: string, country: string, country_id: int|null, email: null|string, gender: null|string, graduation_date: false|null|string, gravatar_92: string, hide_problem_tags: bool, is_private: bool, locale: string, name: string, preferred_language: null|string, rankinfo: array{name: string, problems_solved: int, rank: int, status: string}, scholar_degree: null|string, school: null|string, school_id: int|null, state: null|string, state_id: int|null, username: string, verified: bool}}}, template: string}
      */
-    public static function getProfileDetailsForSmarty(
-        \OmegaUp\Request $r,
-        bool $isProfileEdit = false,
-        bool $isUserEmailEdit = false,
-        bool $isResults = false
-    ) {
-        $smartyProperties = [
+    public static function getProfileDetailsForSmarty(\OmegaUp\Request $r) {
+        return [
+            'smartyProperties' => self::getProfileDetails($r),
+            'template' => 'user.profile.tpl',
+        ];
+    }
+
+    /**
+     * @return array{smartyProperties: array{profile: array{userinfo: array{birth_date: null|string, classname: string, country: string, country_id: int|null, email: null|string, gender: null|string, graduation_date: false|null|string, gravatar_92: string, hide_problem_tags: bool, is_private: bool, locale: string, name: string, preferred_language: null|string, rankinfo: array{name: string, problems_solved: int, rank: int, status: string}, scholar_degree: null|string, school: null|string, school_id: int|null, state: null|string, state_id: int|null, username: string, verified: bool}}, SUPPORTED_LANGUAGES: list<string>, COUNTRIES: list<\OmegaUp\DAO\Countries>}, template: string}
+     */
+    public static function getProfileEditDetailsForSmarty(\OmegaUp\Request $r) {
+        $response = self::getProfileDetails($r);
```

Comment:
no necesitaría esto ser más bien

```suggestion
        $smartyProperties = self::getProfileDetails($r)['smartyProperties'];
```

? si no, no estás estableciendo el elemento correcto.